### DISCUSSION TITLE
Compass Heading Line required for HIKING Organic Maps version

### DISCUSSION THREAD

**Discussion by pwb999** [organicmaps/organicmaps#2686] [Ideas]
I use Locus Maps for walking and find that their option of displaying the current heading derived from the mobiles internal magnetic compass sensor from your current position is really useful in determining what direction you are travelling
<img width="201" alt="Screenshot 2022-06-07 at 13 10 28" src="https://user-images.githubusercontent.com/58628197/172376005-3de2e831-33f1-4ef3-87c7-c47224e5f85b.png">
.In this image the ORANGE line is your Compass Heading direction. The other RED Crosshairs is a line from current GPS position to another location that you choose in order to get its Direction and Distance. Again useful.
If this App is to be a HIKING app, some consideration should be paid to what features hikers need. A Compass and a Map are the basics of all such navigation especially across pathless terrain in thick cloud when you must rely on your Compass to keep you safe. Agreed this need completely different from those who simply want a dedicated Car SatNav device. But they do have at least Road Signs to help them. Theres nothing similar in the wild outdoors to help !!!

**Comment by biodranik** [comment]
The magnetic compass is often wrong in reality, no?

**Comment by pwb999** [comment]
The most common Scenario for hikers is you are on a ridge and the cloud suddenly encloses you. You cannot see where you are going. Take out your Silva/Suunto magnetic compass. It tells you the path you are on is SAFE to continue. You descend that path out of the cloud and thank the stars you purchased a simple magnetic compass. It is the ONLY bit of technology (thanks to Marco Polo) that is reliable in MOST situations. In the case of a good mobile phone its Magnetometer sensor replicates the same thing - it just needs to be Calibrated using a Figure of Eight process that good hikers do anyhow and compare its Heading with other information on the map that is reliable. Some like me carry a table of the Suns Azimuth and its Shadow so that I can if I can see the Sun compare its Bearing with that of my Phone to ensure it has been correctly calibrated. Its standard procedure for many hikers. In addition look at the kit that orienteers use - a simple compass that points NORTH and then they work out their Heading from that - it is very reliable when you understand how to calibrate and compare it with reality ie the map or the planets...<img width="449" alt="Screenshot 2022-06-13 at 19 36 35" src="https://user-images.githubusercontent.com/58628197/173422067-6e9f9a99-0b5d-44de-8fc0-e4eee0ecc9ee.png">
A Satnav can always be wrong - a properly calibrated Compass is rarely wrong.

**Comment by pwb999** [comment]
Sorry. Forgot to add this is how many hikers calibrate the magnetic compass in their phone. I generally try to ensure that the uT reading is about 37 to 40 and always compare its reading with the a Map Feature to confirm its good for navigation. The problem with some phones is that they do not have an internal magnetic compass and have to rely on the GPS heading which only works when you are moving. In many cases having a Heading Line would be really useful when you are stationary and need to make a decision - Do i take the left or right Fork ??
![Screenshot_20220613-194729_Compass](https://user-images.githubusercontent.com/58628197/173424256-886cc144-ae66-4d25-8342-0f730ab83cb3.jpg)

Once again love Organic Maps is excellent and greatly appreciate your Teams hard work. Some additional features as I suggest would make a great recommendation for hikers.

Best wishes
Peter B.
